ui/src/app/services/profile-storage.service.ts

```diff
@@ -0,0 +1,27 @@
+import 'rxjs/Rx';
+
+import {Injectable} from '@angular/core';
+import {ReplaySubject} from 'rxjs/ReplaySubject';
+
+import {Profile} from 'generated';
+import {ProfileService} from 'generated';
+
+
+@Injectable()
+export class ProfileStorageService {
+  private activeCall = false;
+  private profile = new ReplaySubject<Profile>(1);
+  public profile$ = this.profile.asObservable();
```

Comment:
Unless this is following some existing convention, would prefer not to use `$` in the property name. Could be `profile` or `profileObs`?

Reply:
I've read that the `$` is convention to denote an observable in angular (https://cycle.js.org/basic-examples.html#basic-examples-increment-a-counter-what-is-the-convention). I'm still ramping up on the angular world, so this is an odd style for me as well.

Reply:
Yeah, when I was working on the `SignInService` John told me about that convention. I would be happy to follow a different convention if we wanted to use another one.

Reply:
I don't generally like Hungarian notation but we pretty frequently have a source and an observable view on that source (that is the case here in fact), and the dollar suffix is a visible and succinct way to indicate which is which.